mod-source-record-manager-server/src/main/java/org/folio/services/flowcontrol/RawRecordsFlowControlServiceImpl.java

```diff
@@ -22,61 +21,61 @@
-  private Integer maxSimultaneousRecords;
+  @Value("${di.flow.control.max.simultaneous.chunks:2}")
+  private Integer maxSimultaneousChunks;
   @Value("${di.flow.control.records.threshold:25}")
   private Integer recordsThreshold;
   @Value("${di.flow.control.enable:true}")
   private boolean enableFlowControl;
 
   @Autowired
   private KafkaConsumersStorage consumersStorage;
-  @Autowired
-  private EventProcessedService eventProcessedService;
 
   /**
-   * Represents the current state of flow control to make decisions when need to make pause/resume calls.
+   * Represents the current state of flow control to make decisions when need to make new fetch call.
    * It is basically counter, that increasing when DI_RAW_RECORDS_CHUNK_READ event comes and decreasing when
-   * DI_COMPLETE, DI_ERROR come.
+   * DI_COMPLETE, DI_ERROR, duplicated chunks come.
    *
-   * When current state equals or exceeds di.flow.control.max.simultaneous.records - all raw records consumers from consumer's group should pause.
-   * When current state equals or below di.flow.control.records.threshold - all raw records consumers from consumer's group should resume.
+   * When current state below di.flow.control.records.threshold - all raw records consumers from consumer's group should fetch.
    */
   private final Map<String, Integer> currentState = new ConcurrentHashMap<>();
 
+  /**
+   * Represents the previous flow control state to decide when to turn on the consumer to get the next chunk of data,
+   * in case a module instance has not received any messages in 2 minutes (for example, when the second module
+   * instance processed them).
+   */
+  private final Map<String, Pair<Integer, Integer>> historyState = new ConcurrentHashMap<>();
+
   @PostConstruct
   public void init() {
     LOGGER.info("init:: Flow control feature is {}", enableFlowControl ? "enabled" : "disabled");
   }
 
   /**
-   * This method schedules resetting state of flow control. By default it triggered each 5 mins.
+   * This method schedules resetting state of flow control. By default, it triggered each 2 mins.
    * This is distributed system and count of raw records pushed to process can be not corresponding
    * with DI_COMPLETE/DI_ERROR, for example because of imports are stuck.
    *
-   * For example, batch size - 10, max simultaneous - 20, records threshold - 10.
-   * Flow control receives 20 records and pause consumers, 5 DI_COMPLETE events came, 15 was missed due to some DB issue.
-   * Current state would equals to 15 and threshold will never met 10 records so consumers will be paused forever
+   * For example, batch size - 50 records, max simultaneous - 1 batch, records threshold - 25.
+   * Flow control receives 50 records (1 batch) and pause consumers, 20 DI_COMPLETE events came, 30 was missed due to some DB issue.
+   * Current state would equals to 30 and threshold will never met 24 records so consumers will be paused forever
```

Comment:
`threshold will never met 24 records` should this be 25 instead since your records threshold is set to 25?